src/main/kotlin/io/provenance/client/gas/prices/UrlGasPrices.kt

```diff
@@ -0,0 +1,35 @@
+package io.provenance.client.gas.prices
+
+import com.google.gson.Gson
+import cosmos.base.v1beta1.CoinOuterClass
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.impl.client.HttpClientBuilder
+import java.io.InputStreamReader
+
+/**
+ * When provided with a url, fetches an object of shape '{"gasPrice":nnn,"gasPriceDenom":"denom"}'
+ */
+open class UrlGasPrices(private val uri: String) : GasPrices {
+    private val client = HttpClientBuilder.create().build()
+    private val gson = Gson().newBuilder().create()
+
+    private data class GasPrice(val gasPrice: Int, val gasPriceDenom: String) {
+        fun toCoin(): CoinOuterClass.Coin = CoinOuterClass.Coin.newBuilder()
+            .setAmount(gasPrice.toString())
+            .setDenom(gasPriceDenom)
+            .build()
+    }
+
+    override fun invoke(): CoinOuterClass.Coin {
+        val result = client.execute(HttpGet(uri))
+        require(result.statusLine.statusCode in 200..299) {
```

Comment:
do we want to have a (optional, use if supplied) fallback default price to use in the event that the uri is unreachable?

Reply:
it will fall back to the cached transaction (if caching is enabled). what would we fall back to? (semi-rhetorical question, i honestly don't know what a good "fallback" value would be).